packages/editor/src/dataviews/actions/utils.ts

```diff
@@ -12,11 +12,19 @@ import {
 	TEMPLATE_POST_TYPE,
 } from '../../store/constants';
 
-import type { Post, TemplateOrTemplatePart } from '../types';
+import type { Post, TemplatePart, Template } from '../types';
+
+export function isTemplate( post: Post ): post is Template {
+	return post.type === TEMPLATE_POST_TYPE;
```

Comment:
The reason for the `is` it's because can be also a Pattern type which doesn't have the `type` prop?

Reply:
if you do `if ( isTemplate( something ) )` Typescript will know that within the condition we have the `Template` type for `something` so it validates all the properties... properly. Without it, typescript will tell you for instance that something doesn't have a property that exists only in `Template` but not for generic `BasicPost`